src/secrets/aws_secretsmanager.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::{HashMap, HashSet};
+
+use aws_sdk_secretsmanager::{config, Client};
+use futures::executor;
+use vector_lib::configurable::{component::GenerateConfig, configurable_component};
+
+use crate::aws::{create_client, AwsAuthentication, ClientBuilder, RegionOrEndpoint};
+use crate::config::ProxyConfig;
+use crate::tls::TlsConfig;
+use crate::{config::SecretBackend, signal};
+
+// TODO: should this be moved to a different package, maybe aws or common?
+pub(crate) struct SecretsManagerClientBuilder;
+
+impl ClientBuilder for SecretsManagerClientBuilder {
+    type Client = Client;
+
+    fn build(config: &aws_types::SdkConfig) -> Self::Client {
+        let config = config::Builder::from(config).build();
+        Client::from_conf(config)
+    }
+}
+
+/// Configuration for the `aws_secretsmanager` secrets backend.
+#[configurable_component(secrets("aws_secretsmanager"))]
+#[derive(Clone, Debug)]
+pub struct AwsSecretsmanagerBackend {
+    /// ID of the secret to resolve.
+    pub secret_id: String,
+
+    #[serde(flatten)]
+    #[configurable(derived)]
+    pub region: RegionOrEndpoint,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub auth: AwsAuthentication,
+
+    #[configurable(derived)]
+    pub tls: Option<TlsConfig>,
+}
+
+impl GenerateConfig for AwsSecretsmanagerBackend {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(AwsSecretsmanagerBackend {
+            secret_id: String::from("secret-id"),
+            region: Default::default(),
+            auth: Default::default(),
+            tls: None,
+        })
+        .unwrap()
+    }
+}
+
+impl SecretBackend for AwsSecretsmanagerBackend {
+    fn retrieve(
+        &mut self,
+        secret_keys: HashSet<String>,
+        _: &mut signal::SignalRx,
+    ) -> crate::Result<HashMap<String, String>> {
+        let client: Client = executor::block_on(async {
+            return create_client::<SecretsManagerClientBuilder>(
+                &self.auth,
+                self.region.region(),
+                self.region.endpoint(),
+                &ProxyConfig::default(),
+                &self.tls,
+            )
+            .await;
+        })?;
+
+        let get_secret_value_response = executor::block_on(async {
+            return client
+                .get_secret_value()
+                .secret_id(&self.secret_id)
+                .send()
+                .await;
+        })?;
```

Comment:
Just do this in a single `block_on` call.

Reply:
yeah, i actually tried to do that but didn't manage to resolve the error (i'm completely new to rust):
```
error[E0277]: `?` couldn't convert the error to `SdkError<GetSecretValueError, aws_smithy_runtime_api::http::Response>`
  --> src/secrets/aws_secrets_manager.rs:68:19
   |
68 |             .await?;
   |                   ^ the trait `std::convert::From<Box<dyn snafu::Error + std::marker::Send + std::marker::Sync>>` is not implemented for `SdkError<GetSecretValueError, aws_smithy_runtime_api::http::Response>`, which is required by `std::result::Result<GetSecretValueOutput, SdkError<GetSecretValueError, aws_smithy_runtime_api::http::Response>>: FromResidual<std::result::Result<std::convert::Infallible, Box<dyn snafu::Error + std::marker::Send + std::marker::Sync>>>`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the trait `std::convert::From<aws_sdk_sqs::error::BuildError>` is implemented for `SdkError<GetSecretValueError, aws_smithy_runtime_api::http::Response>`
   = help: for that trait implementation, expected `aws_sdk_sqs::error::BuildError`, found `Box<dyn snafu::Error + std::marker::Send + std::marker::Sync>`
   = note: required for `std::result::Result<GetSecretValueOutput, SdkError<GetSecretValueError, aws_smithy_runtime_api::http::Response>>` to implement `FromResidual<std::result::Result<std::convert::Infallible, Box<dyn snafu::Error + std::marker::Send + std::marker::Sync>>>`
```
it's probably something really stupid :D

Reply:
Ah yeah, I see what you mean... my bad, should have noticed that it wasn't going to be able to handle those two different error return types.

The way you had it before is fine. 👍🏻

Reply:
i'll revert it tomorrow